services/retrieve_and_rank/v1.js

```diff
@@ -143,7 +143,14 @@ module.exports = function (RED) {
 
         var ranker_id = config.rankerid;
         var question = msg.payload;
-        var query = qs.stringify({q: question, ranker_id: ranker_id, fl: 'id,title'});
+
+        var query;
+
+        if(config.searchmode === 'search') {
+          var query = qs.stringify({q: question, fl: 'id,title'});
+        } else {
+          var query = qs.stringify({q: question, ranker_id: ranker_id, fl: 'id,title'});
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'query' is already declared in line 150.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2404865885/issues/source?bid=3225115&fileBranchId=3233374#l152)